sauce/features/accounts/easy-transaction-approval/index.js

```diff
@@ -32,13 +32,13 @@ export class EasyTransactionApproval extends Feature {
   }
 
   addBudgetVersionIdObserver() {
-    var parent = this;
+    var _this = this;
 
     var applicationController = ynabToolKit.shared.containerLookup('controller:application');
     applicationController.addObserver('budgetVersionId', function () {
       Ember.run.scheduleOnce('afterRender', this, function () {
-        parent.initKeyLoop = true;
-        parent.initClickLoop = true;
+        _this.initKeyLoop = true;
+        _this.initClickLoop = true;
       });
     });
   }
```

Comment:
You can use the built-in route observer instead of rolling your own, Like this;
```
  onRouteChanged() {
    if (!this.shouldInvoke()) { return; }
    this.initKeyLoop = true;
    this.initClickLoop = true;
  }
```

Reply:
Does that actually do the same thing though? I'm wanting to reset those values if the budget is changed, not if the route is changed?

Reply:
What’s being observed is the same. Look at routeChangeListener.js. Are you saying that budgetVersionId changes each time something in a budget changes? I thought that was the internal ID of the budget and changed when the user switches to a completely different budget.

Reply:
Sorry, I think we're meaning the same thing? I wanted the feature to reset (to re-call the methods that watch for key entry and right clicks) if the user's selected budget file changed (not the budget section itself).

My understanding is that onRouteChanged() watches for moving between sections of the same budget file (Accounts, Budget, Reports), and doesn't necessarily watch for the budget file itself changing?

Reply:
My mistake. I meant you could use onBudgetChanged().

Reply:
Oh, got it. So same example, just different method name? 

onBudgetChanged() {
    if (!this.shouldInvoke()) { return; }
    this.initKeyLoop = true;
    this.initClickLoop = true;
  }

Thanks.

Reply:
Yes.